### DISCUSSION TITLE
Doubt about Microsoft Community Promise

### DISCUSSION THREAD

**Discussion by federico-r-figueredo** [dotnet/runtime#74924] [General] [answered]
I think that I could be missing something but, why would this runtime repository 've a copy of [Microsoft's community promise](https://github.com/dotnet/runtime/blob/main/PATENTS.TXT)? This project was born open source under FOSS license since day 1, why would it need this CP? It's not like it's an implementation of closed source tech that has not been standardized (as mono project did w/ proprietary & closed source .NET Framework). I didn't saw the same document in [roslyn ](https://github.com/dotnet/roslyn)or [aspnetcore ](https://github.com/dotnet/aspnetcore)repositories. If there's a valid explanation for that, wouldn't those repositories need also a copy of the CP?

**Comment by richlander** [comment, marked as answer]
This all seems like ancient history now. I totally get why the document seems odd.

As we were getting ready to announce .NET Core 1.0, we wanted to remove FUD that we thought might get raised up. What might that have been? If you do some historical research, you'll find that there was (at one point) deep concern in the Linux community about adding the Mono project to official archives for Linux distributions. This was based on the idea that Microsoft might assert IP rights in unfavorable ways. In short, Mono might be a sort of legal trojan horse. Let's be clear. That never happened and the horse was just a mild-mannered pony. However, I sympathize with the concern. Microsoft discourse on Linux during that same time period was bad.

We could have just gone with a "our warranty is good" sort of approach to build up community confidence. Instead, we published a clear and binding statement not to assert .NET patents. From what I can tell, this approach served its purpose. .NET has been significantly embraced on Linux, which is great. The change in company discourse likely helped significantly, too.

We scoped the promise to the runtime and libraries since they were a fork of .NET Framework and covered the same surface area as Mono. ASP.NET Core and other components in .NET Core 1.0 were new from-scratch projects so didn't seem important to cover. Other projects, like Windows Forms and WPF, came later and we never revisited the scope of the community promise. To a large degree, the time and need for the promise has passed. It comes up very rarely at this point. That said, the promise remains in place and the statement/file will remain in the dotnet/runtime repo forever.

**Reply by MetalMaxMX** [reply]
I have a few questions about this actually. 

1) If you wanted to remove the FUD and the uncertainty on the Linux communities. Wouldn't have been more proper to release a patent-grant to all C# or C#-related patents? Similar to what other programming languages already do (Specially those that use an Apache license) 

2) How is the promise binding? If, say, Microsoft decides to change leadership at a later time and start potentially threatening, be it directly or indirectly other communities with patents. What stops Microsoft from not asserting its patents rights to other users? You may think that I am being paranoid but after the news came out of what a certain game engine company is doing, do allow me to a bit cautious. 

3) Finally, does the patent promise apply to all .NET repositories? Do all of them also have the PATENTS.TXT tucked in somewhere? 

I have seen interest in C# growing here and there among some communities. But it is worrying to me about the potential liabilities that patents may hold, which could stop some people from contributing promising packages written for this. And that's another reason why I am asking. 

Cheers!
MetalMaxMX.

**Reply by BoyFaceGirl** [reply]
Since. net has been open source for 10 years now, does Microsoft have any disputes over fees or closed source for patent open source agreements? So what have you been doing in these 10 years? Are you feeling scared every day? Now, free and open source, just do what you need to do now. Use it and use your knowledge to make your own contribution to. Net instead of conspiracy theories everywhere. If you have contributed enough valuable code, even if Microsoft changes its focus on closed source,  they still need to ask yourself if you agree. Do less meaningless satirical worries, and contribute more!!!!

**Reply by MetalMaxMX** [reply]
> Since. net has been open source for 10 years now, does Microsoft have any disputes over fees or closed source for patent open source agreements? So what have you been doing in these 10 years? Are you feeling scared every day? Now, free and open source, just do what you need to do now. Use it and use your knowledge to make your own contribution to. Net instead of conspiracy theories everywhere. If you have contributed enough valuable code, even if Microsoft changes its focus on closed source, they still need to ask yourself if you agree. Do less meaningless satirical worries, and contribute more!!!!


Then, I'll ask you. Why is Microsoft not releasing these patents into the public domain? What are they losing out on if they were to do that? This is a legit question I am asking to you. Nearly every other mainstream programming language of notice (C, C++, Rust, Python, Javascript, Swift, etc.) has no issues with this because either they're a full ISO standard (like C and C++) or their license gives the user a patent grant (Swift with Apache License for example), they don't need a " community promise", you are free with those languages.

Big corporations have a history of turning against their users due to any change of leadership or management, and as you said, if MS changes its focus towards a closed-source environment, there is risk of forking .NET itself because MS will have patents to combat those forks.

I agree with you that we shouldn't be afraid to develop software we like due to patents, I personally consider them an absurdity on software. But if there is possibility of lessening the risk towards the end-user and developers of them, then I think that would make for a better ecosystem overall.

**Reply by BoyFaceGirl** [reply]
Microsoft has spent the most money and various costs, and has the right to do anything? If you invest more than Microsoft, you can do whatever you want. Why must we struggle with Microsoft? He is originally a commercial company, and many decisions are also based on commercial considerations. You can try using Java. If you are not happy with Microsoft's approach, just switch platforms. What means did Microsoft use to force you to use it? Microsoft needs you to use c #? It's better to think about how you can contribute to open source instead of spending all day on conspiracy theory. Do meaningful things, don't be a cyber junkie conspiracy theorist!

**Reply by MetalMaxMX** [reply]
So, because Microsoft spent money on it, that means they are allowed to do anything? I mean, you said so yourself, this could include changing the terms and actively going against their users in the future, but it is fine because they spent money on it! 

I'll give you the point though that you're not forced to use .NET or C#, Java is a pretty nice alternative all things considered. Though I am not in agreement about contributing to .NET due to non-free patents, to open-source in general though is still a nice act. Do allow me to end that, given how corporations are currently behaving and all, is never a bad idea to act cautious about them, even if one gets labeled as a "cyber junkie conspiracy theorist" :-)

Cheers.

**Reply by BoyFaceGirl** [reply]
Okay, you go use Java quickly, leave the. Net community quickly, leave forever, and enjoy using Java You will definitely be devastated by your departure. You are the metaverse, and even without you, the universe will not continue to operate.
